runtime/rasdump/dmpsup.c

```diff
@@ -1365,7 +1366,42 @@ initSystemInfo(J9JavaVM *vm)
 			}
 		}
 	}
-	appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_PATTERN, J9RAS_CORE_PATTERN_FILE);
+	{
+		J9RASSystemInfo *corePatternInfo = appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_PATTERN, J9RAS_CORE_PATTERN_FILE);
+		if (NULL != corePatternInfo) {
+			/* A common core_pattern is Dynatrace; for example, |/opt/dynatrace/oneagent/agent/rdp
+			 * This program sends the core to the originally configured
+			 * core_pattern as storied in, for example,
+			 * /opt/dynatrace/oneagent/agent/conf/original_core_pattern
+			 *
+			 * If we find this Dynatrace core_pattern, extract its installation
+			 * directory and then read original_core_pattern relative to that.
+			 */
+			const char *corePattern = (const char *)corePatternInfo->data;
+			if ('|' == corePattern[0]) {
+				char *dynatracePath = strstr(corePattern, "/oneagent/agent/rdp");
+
+				/* Check if core_pattern includes the Dynatrace agent and sanity check
+				 * that the length of core_pattern is smaller than our buffer.
+				 */
+				if ((NULL != dynatracePath) && (strlen(corePattern) < MAX_INTERESTING_LENGTH)) {
+					char namebuf[MAX_INTERESTING_LENGTH];
+					/* The length of the agent path prefix, minus the pipe character. */
+					size_t prefixLength = dynatracePath - corePattern - 1;
+
+					/* Copy starting after the pipe character. */
+					strncpy(namebuf, corePattern + 1, prefixLength);
+					namebuf[prefixLength] = '\0';
+
+					/* Append the location of original_core_pattern. */
+					strncat(namebuf, "/oneagent/agent/conf/original_core_pattern", sizeof(namebuf) - prefixLength - 1);
+
+					/* Finally, read the file contents. */
+					appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_ORIGINAL_PATTERN, namebuf);
+				}
```

Comment:
* the constructed file name may not fit in the buffer because the added suffix is longer than the removed segment
* `memcpy()` is preferred over `strncpy()` when the length is known

I suggest:
```c
				static const char search[] = "/oneagent/agent/rdp";
				static const char replacement[] = "/oneagent/agent/conf/original_core_pattern";
				const char *dynatracePath = strstr(corePattern, search);
				char namebuf[MAX_INTERESTING_LENGTH];

				/* Check if core_pattern includes the Dynatrace agent and ensure
				 * that the original core_pattern path will fit in our buffer.
				 */
				if ((NULL != dynatracePath)
				&& (strlen(corePattern + 1) < (sizeof(namebuf) - (sizeof(replacement) - sizeof(search))))
				) {
					/* The length of the agent path prefix, minus the pipe character. */
					size_t prefixLength = dynatracePath - corePattern - 1;

					/* Copy starting after the pipe character. */
					memcpy(namebuf, corePattern + 1, prefixLength);

					/* Append the location of original_core_pattern. */
					strcpy(namebuf + prefixLength, replacement);

					/* Finally, read the file contents. */
					appendSystemInfoFromFile(vm, J9RAS_SYSTEMINFO_CORE_ORIGINAL_PATTERN, namebuf);
				}
```

Reply:
Thank you